docs/guides/getting-started.md

```diff
@@ -0,0 +1,140 @@
+# Getting Started
+
+You can create a `react-sketchapp` project with `skpm`, by cloning a ready-made [example](../examples.md), or by manually setting up the `package.json` and `manifest.json` scripts (advanced usage).
+
+## Environment Setup
+
+You will need npm, Node and Sketch.
+
+- Code editor e.g. [VSCode](https://code.visualstudio.com/), [Atom](https://atom.io/)
+- Node.js & `npm` – [install with Homebrew](https://nodejs.org/en/download/package-manager/#macos) (or install with [Node Version Manager](https://nodejs.org/en/download/package-manager/#nvm))
+- [Sketch](https://www.sketch.com/)
+  - requires macOS
+
+## Creating a Project With Skpm
+
+**Replace** `my-app` with your desired project name:
+
+### Installation
+
+```bash
+npm i -g skpm
+skpm create my-app --template=airbnb/react-sketchapp # template is a GitHub repo
+cd my-app
+```
+
+### Setup
+
+You can now open `my-app` in your code editor of choice. You will see a `src` folder with a `manifest.json` file and Sketch entrypoint (e.g. `my-command.js`). If you wish to rename `my-command.js`, you can do so and update the file name in `script` in `manifest.json`
+
+Example modifications (assuming we want to rename the entrypoint file to `main.js` and don't want to have sub-commands):
+
+`src/manifest.json`
+
+```diff
+  "commands": [
+    {
+-      "name": "my-command",
++      "name": "My App Name: Sketch Components"
+-      "identifier": "my-command-identifier",
++      "identifier": "main",
+-      "script": "./my-command.js"
++      "script": "./main.js"
+    }
+  ],
+  "menu": {
+-    "title": "my-app",
+-    "items": [
+-      "my-command-identifier"
+-    ]
++    "isRoot": true,
++    "items": [
++      "main"
++    ]
++  }
+  }
+```
+
+### Rendering to Sketch
+
+To render your app to Sketch, open the Sketch application, create a new blank document, then go to your Terminal and run:
+
+```bash
+// Make sure you've already done `cd my-app`
+npm run render
+```
+
+If you forget to create a new document, you may get an error saying `null is not an object`. You can handle this case with the instructions below:
```

Comment:
👍 Is this not worth creating a new Sketch document by default in the render function? If no documents are found.

e.g. https://github.com/airbnb/react-sketchapp/blob/9ed55801f242d954e8618da343310ec22a293ce1/src/utils/getDocument.ts#L3-L8

->

```js
 export const getDocumentDataFromContext = (ctx: SketchContext): SketchDocumentData => {
  try {
    return ( 
     ctx.document || 
     (ctx.actionContext || {}).document || 
     NSDocumentController.sharedDocumentController().currentDocument() 
   ).documentData(); 
  } catch(err) {
    if (sketch.documents.length === 0) {
      return new sketch.Document().sketchObject.documentData();
    }
  }

```

Reply:
No I don't think it should do something that it wasn't explicitly told not to. If we have a nice error message with some suggestion to fix the issue, it will be a lot more actionable than if we sometimes create a new doc but not always

Reply:
Should I open up a PR for:

```js
 export const getDocumentDataFromContext = (ctx: SketchContext): SketchDocumentData => {
  try {
    return ( 
     ctx.document || 
     (ctx.actionContext || {}).document || 
     NSDocumentController.sharedDocumentController().currentDocument() 
   ).documentData(); 
  } catch(err) {
    if (err.includes('null is not an object')) {
      throw new Error('No open document found. Please try opening/creating a document, or you could use \'render(<App />, sketch.getSelectedDocument() || new sketch.Document())\'');
    }
  }
```

Reply:
Let's avoid the plugin sketch API for now (because it might be a pain on nodejs and not working on sketch <49).

But I think it can be just
```js
export const getDocumentDataFromContext = (ctx: SketchContext): SketchDocumentData => {
  let document = ctx.document || 
     (ctx.actionContext || {}).document || 
     NSDocumentController.sharedDocumentController().currentDocument()
  if (!document || !document.documentData) {
    throw new Error('No open document found.\nTry opening/creating a document before calling `render`.\n\nFor more information, check out http://airbnb.io/react-sketchapp/docs/guides/rendering.html');
  }
}
```